ui/src/components/debug_logger/mod.rs

```diff
@@ -34,49 +64,179 @@ pub fn DebugLogger(cx: Scope) -> Element {
+                                    filter_level.set(Level::Trace);
+                                },
+                                tooltip: cx.render(rsx!(
+                                    Tooltip {
+                                        arrow_position: ArrowPosition::Top,
+                                        text: "Trace".into()
+                                    }
+                                )),
+                            },
+                        }
+                    })),
+                    Button {
+                        text: "State".into(),
+                        icon: Icon::Square3Stack3d,
+                        appearance: if active_tab.get() == &Tab::State { Appearance::Primary } else { Appearance::Secondary },
+                        onpress: |_| {
+                            active_tab.set(Tab::State);
+                        }
+                    },
+                    Button {
+                        text: "Web Inspector".into(),
+                        icon: Icon::ArrowTopRightOnSquare,
+                        appearance: Appearance::Secondary,
+                        onpress: |_| {
+                            window.webview.open_devtools();
+                        }
+                    },
+                },
+                div {
+                    class: "logger-nav-right",
+                    aria_label: "debug-logger-nav-right",
+                    IconElement {
+                        icon: if state.read().ui.theme.clone().unwrap_or_default().name == "Light" {
+                            Icon::Sun
+                        } else {
+                            Icon::Moon
+                        }
+                    },
+                    Switch {
+                        active: state.read().ui.theme.clone().unwrap_or_default().name == "Light",
+                        onflipped: move |_| {
+                            let current_theme = state.read().ui.theme.clone().unwrap_or_default();
+
+                            if current_theme.name != "Light" {
+                                let light_theme = get_available_themes().iter().find(|t| t.name == "Light").unwrap().clone();
+                                state.write().mutate(Action::SetTheme(Some(light_theme)));
+                            } else {
+                                state.write().mutate(Action::SetTheme(None));
                             }
-                        )
-                    })
-                }
+                        }
+                    }
+                },
+            },
+            match active_tab.get() {
+                Tab::Logs => rsx!(div {
+                    aria_label: "debug-logger-body",
+                    class: "body",
+                    div {
+                        class: "body-scroll",
+                        logs_to_show.iter().filter(
+                            |&x| &x.level == filter_level.get() || filter_level.get() == &Level::Debug
```

Comment:
This is more of a nik pik, but if `Level` derives `Copy` you can dereference it instead. 

```suggestion
                            |&x| x.level == *filter_level.get() || *filter_level.get() == Level::Debug
```